File: DailyQuestion/java/55. Jump Game.java
package DailyQuestion.java;
class Solution {
    public boolean canJump(int[] nums) {
        if(nums == null || nums.length==0){return false;}
        if(nums.length == 1){return true;}
        int count = 0;
        int len = nums.length-1;
        for(int i=0; i<len;i++){
            count = Math.max(--count, nums[i]);
            if(count == 0){break;}
            if(nums[i] + i >= len){
                return true;
            }
        }
        return false;
    }
}

File: DailyQuestion/js/55. Jump Game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let count=0;
    let len = nums.length-1;
    if(len === 0){return true;}
    for(let i=0; i<len; i++){
        if((count=Math.max(--count,nums[i])) <=0){
            break;
        }
        if(nums[i] + i >= len){
            return true;
        }
    }
    return false;
};

console.log(canJump([2,0,0]))